### ISSUE TITLE
Deferred.then() documentation needs more emphasis on chaining of Deferreds/Promises

### ISSUE THREAD

**Issue by EarthCitizen** [jquery/api.jquery.com#638]
The documentation for Deferred.then() states:

> These filter functions can return a new value to be passed along to the promise's .done() or .fail() callbacks, or they can return another observable object (Deferred, Promise, etc) which will pass its resolved / rejected status and values to the promise's callbacks.

The fact that the filter functions can return a new Deferred or Promise, and the importance of that is **greatly** under emphasized. The way it is written seems like a small unimportatnt side note, but important functionality is not clear due to the sublety of the statement. And that functionality being that you can, for example, chain together multiple functions (returning Promises) which will not execute until the previous promise has been resolved.

Let me give an example of why this feature needs to be more strongly emphasized:

I recently ran into a situation where I was displaying validation errors and "OK, Cancel"-style dialogs using jQuery-ui dialogs. As we all know, the callbacks for the dialogs are asynchronous. Therefore, I needed further validation to wait until the user had clicked a button on one of the dialogs, and depending on the action taken by the user, halt all remaining validation.

The answer was to break down the validation into logical sections/functions, and each validation function would create its own Deferred and return a Promise. Then, the jQuery-ui dialogs gets a callback function for the buttons that either resolve or reject the Deferred.

So, basically, you can chain together validation functions which display jQuery-ui dialogs and have them queued as if they were synchronous, and skipping the execution of other validation if one of them rejects its Deferred.

The general use case being:

```JavaScript
$.when(validationOneReturningPromise())
.then(validationTwoReturningPromise)
.then(validationTwoReturningPromise)
.done(function() { someForm.submit(); });
```

This is only one scenario out of many possible that could greatly benefit from clarification and stronger emphasis in this area of the documentation. The features available with `when()` and `then()` and quite powerful, but the full extent of the possible uses is, I feel, undersold and underdemonstrated in the docs.

Thank You

**Comment by dmethvin** [comment]
I agree this page needs some rework, and it should get some when we release 3.0 which supports Promise/A+ compliance for the `.then()` method. I don't think we should be getting into detailed how-to information here, that would be better done in https://github.com/jquery/learn.jquery.com with a feature-style article. We should have a beta of 3.0 ready soon and would love to have you try it. If you're interested in writing an article at the learn site that would be great as well!

**Comment by EarthCitizen** [comment]
Yes, I certainly was not suggesting the details I gave be put into the documentation, but rather giving an example of something great that can be done, but is not clear from reading the doc.

I think updating the doc with just a couple of sentences kind of saying in a general way that the kind of thing I was referring to is possible. That is, chaining promises in stages and that when-then are not only for ajax.
